### PR TITLE
chore: Validate with ruff

### PR DESCRIPTION
**Pull request by RakuJa**
# Description

Implement ruff code guidelines

## Type of change

Please delete options that are not relevant or put an 'x' on the desired options.

- [x] New feature (non-breaking change which adds functionality)
# Checklist:

This checklist is used as a reminder to avoid half-baked code
- [x] The PR title follows the following structure:" type[optional scope]: PREV-XX - LONG DESCRIPTION "
- [x] I have bumped both the PKGBUILD version and controller version, and they are the same.
- [x] I have correctly installed and enabled pre-commit
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] If I have introduced new dependencies I have added them to the pre-commit unittest additional_dependencies AND to the THIRDPARTIES file

### CODE REVIEW

File: app/core/resources/data_validator.py [hunk 1]
```diff
@@ -104,18 +101,21 @@ class DocumentPagesMetadataModel(BaseModel):
...
         last_page = field_values.get("last_page", 0)
         if first_page >= 1 and (first_page <= last_page or last_page == 0):
             return field_values
-        else:
-            raise HTTPException(
-                status_code=422, detail=message.NUMBER_OF_PAGES_NOT_VALID
-            )
+
+        raise HTTPException(
+            status_code=422,
```
Comment (federicorispo) [inline app/core/resources/data_validator.py]:
```suggestion
            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
```

File: app/core/routers/document.py [hunk 2]
```diff
@@ -102,12 +103,13 @@ async def post_preview(
...
             media_type="application/pdf",
-        )
+        ),
     )
 
 
 @router.post(
-    "/{area}/thumbnail/", responses={400: {"description": message.INPUT_ERROR}}
+    "/{area}/thumbnail/",
+    responses={400: {"description": message.INPUT_ERROR}},
```
Comment (federicorispo) [inline app/core/routers/document.py]:
```suggestion
    responses={status.HTTP_400_BAD_REQUEST: {"description": message.INPUT_ERROR}},
```

Comment (federicorispo) [reply app/core/routers/document.py]: See also row 167

File: app/core/routers/pdf.py [hunk 3]
```diff
@@ -101,7 +101,8 @@ async def post_preview(
 
 
 @router.post(
-    "/{area}/thumbnail/", responses={400: {"description": message.INPUT_ERROR}}
+    "/{area}/thumbnail/",
+    responses={400: {"description": message.INPUT_ERROR}},
```
Comment (federicorispo) [inline app/core/routers/pdf.py]:
```suggestion
    responses={status.HTTP_400_BAD_REQUEST: {"description": message.INPUT_ERROR}},
```

File: app/core/services/document_manipulation/document_manipulation.py [hunk 4]
```diff
@@ -154,7 +161,7 @@ def convert_pdf_to_image(
         )
     except pypdfium2.PdfiumError as e:
         log.info(f"Wrong pdf file passed, error: {e}")
-        raise HTTPException(status_code=400, detail="Invalid pdf file")
+        raise HTTPException(status_code=400, detail="Invalid pdf file") from e
```
Comment (federicorispo) [inline app/core/services/document_manipulation/document_manipulation.py]:
```suggestion
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Invalid pdf file") from e
```

File: app/core/services/document_service.py [hunk 5]
```diff
@@ -47,17 +52,19 @@ async def retrieve_doc_and_create_preview(
                     first_page_number=first_page_number,
                     last_page_number=last_page_number,
                     content=io.BytesIO(
-                        response_data.value_or(RequestResp(status_code=200)).content
+                        response_data.value_or(RequestResp(status_code=200)).content,
```
Comment (federicorispo) [inline app/core/services/document_service.py]:
```suggestion
                        response_data.value_or(RequestResp(status_code=status.HTTP_200_OK)).content,
```

File: app/core/services/document_service.py [hunk 6]
```diff
@@ -100,22 +111,24 @@ async def retrieve_doc_and_create_thumbnail(
...
-        response_data=response_data
+        response_data=response_data,
     )
     return response_error.map(FastApiResp).value_or(
         FastApiResp(
             content=(
                 await document_manipulation.convert_file_to(
                     content=io.BytesIO(
-                        response_data.value_or(RequestResp(status_code=200)).content
+                        response_data.value_or(RequestResp(status_code=200)).content,
```
Comment (federicorispo) [inline app/core/services/document_service.py]:
```suggestion
                        response_data.value_or(RequestResp(status_code=status.HTTP_200_OK)).content,
```

File: app/core/services/image_manipulation/gif_utility_functions.py [hunk 7]
```diff
@@ -47,16 +48,16 @@ def parse_to_valid_gif(
...
         log.debug(f"Content could not be parsed as a valid GIF file: {e}")
     except AttributeError as e:
         log.debug(
             f"Content was parsed with success (valid image), "
-            f"but raised an exception while checking if animated: {e}"
+            f"but raised an exception while checking if animated: {e}",
         )
-    raise ValueError("Content is not a valid GIF")
+    msg = "Content is not a valid GIF"
+    raise ValueError(msg)
```
Comment (federicorispo) [inline app/core/services/image_manipulation/gif_utility_functions.py:60]: What is the benefit of the `msg` variable?

File: app/core/services/image_manipulation/gif_utility_functions.py [hunk 8]
```diff
@@ -257,13 +272,13 @@ def add_circle_margins_to_gif(gif: Image.Image) -> Image.Image:
     size = gif.size
     mask = Image.new("L", size, 255)
     draw = ImageDraw.Draw(mask)
-    draw.ellipse((0, 0) + size, fill=0)
+    draw.ellipse((0, 0, *size), fill=0)
```
Comment (federicorispo) [inline app/core/services/image_manipulation/gif_utility_functions.py:275]: Why this change? I notice that the [#ellipse](https://pillow.readthedocs.io/en/stable/reference/ImageDraw.html#PIL.ImageDraw.ImageDraw.ellipse) method wants a sequence of two values

File: app/core/services/image_service.py [hunk 9]
```diff
@@ -58,7 +59,7 @@ async def retrieve_image_and_create_thumbnail(
             func=_select_thumbnail_module,
         )
     except ValueError as e:
-        raise HTTPException(status_code=400, detail=str(e))
+        raise HTTPException(status_code=400, detail=str(e)) from None
```
Comment (federicorispo) [inline app/core/services/image_service.py]:
```suggestion
        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e)) from None
```

**Review by federicorispo — CHANGES_REQUESTED** [review]
After a quick grep I found these hardcoded status codes. I already commented some of them during the review but I am not sure if the proposed solutions are working or not:

```
app/core/services/document_service.py:55:                        response_data.value_or(RequestResp(status_code=200)).content,
app/core/services/document_service.py:127:                        response_data.value_or(RequestResp(status_code=200)).content,
app/core/services/image_service.py:148:                    response_data.value_or(RequestResp(status_code=200)).content,
app/core/services/pdf_service.py:53:                    response_data.value_or(RequestResp(status_code=200)).content,
app/core/services/pdf_service.py:122:                        response_data.value_or(RequestResp(status_code=200)).content,

app/core/resources/data_validator.py:116:            status_code=422,
app/core/services/document_manipulation/document_manipulation.py:164:        raise HTTPException(status_code=400, detail="Invalid pdf file") from e
app/core/services/image_service.py:62:        raise HTTPException(status_code=400, detail=str(e)) from None
app/core/services/image_service.py:93:        raise HTTPException(status_code=400, detail=str(e)) from None
app/core/services/image_service.py:108:        raise HTTPException(status_code=400, detail=str(e)) from None
app/core/services/image_service.py:123:        raise HTTPException(status_code=400, detail=str(e)) from e
```

File: app/core/services/image_manipulation/gif_utility_functions.py (hunk 8 above)
Comment (RakuJa) [reply app/core/services/image_manipulation/gif_utility_functions.py:275]: (0,0) + size is a tuple concatenation, not a sum of first value of the tuple + first value of the second tuple and so on. => (0,0, size.x, size.y)
So, using the unpacking operator (*) is a little bit more explicit and suggested by default.
(0,0, *size) => (0,0, size.x, size.y)

File: app/core/services/image_manipulation/gif_utility_functions.py (hunk 7 above)
Comment (RakuJa) [reply app/core/services/image_manipulation/gif_utility_functions.py:60]: https://beta.ruff.rs/docs/rules/raw-string-in-exception/

**Review by federicorispo — APPROVED** [review]

**Pull request merged**